// Copyright 2022 Google LLC
//
// Use of this source code is governed by a BSD-style
// license that can be found in the LICENSE file or at
// https://developers.google.com/open-source/licenses/bsd

syntax = "proto2";

package visionai.motion_detection;

message MotionVectorBasedMotionDetectorConfig {
  // Number of spatial grids of the frame from which the features are extracted.
  // Larger this number -> more sensitive to the motion around the frame edges
  optional int32 spatial_grid_number = 1 [default = 5];

  // Buffer frame numbers for temporal smoothing.
  // Larger this number
  //    -> Longer history for temporal smoothing.
  //    -> More time delay on predicting the motion.
  // Smaller this number
  //    -> More sensitive to invisible lighting changes or other noises.
  optional int32 temporal_buffer_frames = 2 [default = 6];

  // Motion vector feature sensitivity
  // Currently used for max magnitude entropy
  optional float motion_sensitivity = 3 [default = 0.6];
}
